source/project/components/app.js

```diff
@@ -0,0 +1,99 @@
+class PopupComponent extends HTMLElement {
+  /**
+   * @constructor
+   * @description Creates an instance of PopupComponent, sets up the shadow DOM, and initializes
+   * the component with CSS and HTML content loaded asynchronously.
+   */
+  constructor() {
+    super();
+    let shadowRoot = this.attachShadow({ mode: "open" });
+
+    // get the css file and append it to the shadow root
+    const style = document.createElement("link");
+    style.rel = "stylesheet";
+    style.href = "styles.css";
+    shadowRoot.append(style);
+
+    // adds the overlay css style to our program(makes the background grey out)
+    // click anywhere outside the container to close the container and remove overlay
+    const overlay = document.createElement("div");
+    overlay.setAttribute("class", "overlay");
+    overlay.addEventListener("click", () => {
+      this.style.display = "none";
+    });
+
+    // waits for the css to load before the html popup occurs
+    const div = document.createElement("div");
+    style.onload = async () => {
+      div.setAttribute("class", "popup-container");
+      const response = await fetch("popup.html");
+      const html = await response.text();
+      div.innerHTML = html;
+    };
+
+    // append everything to the shadow root
+    shadowRoot.append(div, overlay);
+  }
+
+  /**
+   * @method connectedCallback
+   * @description Lifecycle method that is called when the component is inserted into the DOM.
+   * It sets up event listeners for form submission within the shadow DOM.
+   */
+  connectedCallback() {
+    // event listener to form submission
+    setTimeout(() => {
+      this.shadowRoot
+        .querySelector("#journalForm")
+        .addEventListener("submit", this.onSubmit.bind(this));
+    }, 3000);
+
+    // const closeButton = document.getElementById("closeBtn"); old
```

Comment:
In progress, working on getting the close button functionality (same issue that hashim and Matteo had)